core/common/src/main/java/alluxio/AlluxioURI.java

```diff
@@ -499,4 +502,30 @@ public String toString() {
     mUriString = sb.toString();
     return mUriString;
   }
+
+  /**
+   * Computes the hash of this URI, with SHA-256, or MD5, or simple hashCode().
+   *
+   * @return HEX encoded hash string
+   */
+  public String hash() {
+    try {
+      MessageDigest md = MessageDigest.getInstance("SHA-256");
+      md.update(toString().getBytes());
+      return Hex.encodeHexString(md.digest()).toLowerCase();
+    } catch (NoSuchAlgorithmException e) {
+      /* No actions. Continue with other hash method. */
```

Comment:
I don't think it's necessary to consider backup algorithms. 1) any decent platform should have implementation for SHA-256, otherwise they cannot support fundamental things like TLS. 2) if on the client side SHA-256 is available but on the server side is not, the hash will be different for the same URI. This will cause page lookup to fail.

We should stick to only one algorithm, and if SHA-256 isn't available, just throw an exception.

Reply:
“robustness” is to handle different situations as much as possible.
And "don't" throw exception if itself can handle such error.

"any decent platform", actually, this MessageDigest package is from external package.
We are not sure if it will provide such methods. So we may fall back to use Object.hashCode() if all of these are not available.